vendor/github.com/percona/percona-toolkit/src/go/mongolib/fingerprinter/fingerprinter.go

```diff
@@ -120,21 +118,20 @@ func (f *Fingerprinter) Fingerprint(doc proto.SystemProfile) (Fingerprint, error
 		case "group":
 			retKeys = []string{}
 			if g, ok := query.Map()["group"]; ok {
-				if m, ok := g.(bson.M); ok {
-					if f, ok := m["key"]; ok {
-						if keysMap, ok := f.(bson.M); ok {
-							retKeys = append(retKeys, keys(keysMap, []string{})...)
-						}
-					}
-					if f, ok := m["cond"]; ok {
-						if keysMap, ok := f.(bson.M); ok {
-							retKeys = append(retKeys, keys(keysMap, []string{})...)
-						}
-					}
-					if f, ok := m["ns"]; ok {
-						if ns, ok := f.(string); ok {
-							collection = ns
-						}
+				m, err := asMap(g)
```

Comment:
Previously we were ignoring if we can't cast

Reply:
Yes, but I think it was wrong. If the operation is `group`, there must be a `group` field.